iceoryx_hoofs/include/iceoryx_hoofs/internal/posix_wrapper/thread.inl

```diff
@@ -0,0 +1,40 @@
+#ifndef IOX_HOOFS_POSIX_WRAPPER_THREAD_INL
+#define IOX_HOOFS_POSIX_WRAPPER_THREAD_INL
+
+#include "iceoryx_hoofs/posix_wrapper/thread.hpp"
+
+namespace iox
+{
+namespace posix
+{
+inline cxx::expected<ThreadError> ThreadBuilder::create(cxx::optional<Thread>& uninitializedThread,
+                                                        const cxx::function<void()>& callable) noexcept
```

Comment:
If this is intended to be called by the user we need to consider t:
`cxx::function` just as `std::function` is more of a storage type instead of a lightweight "pass-around" type, even if we have a reference here.
For example `std::thread` passes its callable via template parameter (supporting a variadic signature) https://en.cppreference.com/w/cpp/thread/thread/thread, but we do not need this in the same generality. 

We just need to pass something that is invocable with `(void)(void)` signature. This can be done with the following template method```
template< class Function>
inline cxx::expected<ThreadError> ThreadBuilder::create(cxx::optional<Thread>& uninitializedThread, Function &&f);
```

The benefit is that there is no need to construct a `cxx::function` when passing (this will implicitly happen unless our argument is already a `cxx::function`, which generally will not be the case.This also happens if we pass by reference, it will just construct a temporary and pass its reference.

Now this is not completely save, we would need to use `is_invocable_r` with the desired signature. This would make the later check whether the callable is null obsolete as well, since now everything is checked at compile time (another benefit).